x/proof/keeper/session.go

```diff
@@ -69,6 +72,56 @@ func (k msgServer) queryAndValidateSessionHeader(
 	return onChainSession, nil
 }
 
+// validateClaimWindow returns an error if the given session is not eligible for claiming.
+func (k msgServer) validateClaimWindow(
+	ctx context.Context,
+	sessionHeader *sessiontypes.SessionHeader,
+) error {
+	logger := k.Logger()
+
+	sharedParams := k.sharedKeeper.GetParams(ctx)
+
+	// Get the on-chain session end height for the given session header.
+	sessionEndHeight := shared.GetSessionEndHeight(&sharedParams, sessionHeader.GetSessionEndBlockHeight())
+
+	// Get the claim window open and close heights for the given session header.
+	claimWindowOpenHeight := shared.GetClaimWindowOpenHeight(&sharedParams, sessionHeader.GetSessionEndBlockHeight())
+	claimWindowCloseHeight := shared.GetClaimWindowCloseHeight(&sharedParams, sessionHeader.GetSessionEndBlockHeight())
+
+	// Get the current block height.
+	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
+	currentHeight := sdkCtx.BlockHeight()
+
+	// Ensure the current block height is AFTER the claim window open height.
+	if currentHeight < claimWindowOpenHeight {
+		return types.ErrProofClaimOutsideOfWindow.Wrapf(
+			"current block height %d is less than session claim window open height %d",
+			currentHeight,
+			claimWindowOpenHeight,
+		)
+	}
+
+	// Ensure the current block height is BEFORE the claim window close height.
+	if currentHeight > claimWindowCloseHeight {
+		return types.ErrProofClaimOutsideOfWindow.Wrapf(
+			"current block height %d is greater than session claim window close height %d",
+			currentHeight,
+			claimWindowCloseHeight,
+		)
+	}
+
+	logger.
+		With(
+			"current_height", currentHeight,
+			"session_end_height", sessionEndHeight,
+			"claim_window_open_height", claimWindowOpenHeight,
+			"claim_window_close_height", claimWindowCloseHeight,
+		).
+		Debug("validated claim window")
```

Comment:
Thoughts on passing in a pointer to the claim (instead of the session) to `validateClaimWindow` so we can also add the supplier_address as metadata?